README.rst

```diff
@@ -1506,79 +1506,366 @@ Get command details
 
     GET /api/v1/controller/device/{device_id}/command/{command_id}/
 
+List device groups
+##################
+
+.. code-block:: text
+
+    GET /api/v1/controller/group/
+
+Create device group
+###################
+
+.. code-block:: text
+
+    POST /api/v1/controller/group/
+
+Get device group detail
+#######################
+
+.. code-block:: text
+
+    GET /api/v1/controller/group/{id}/
+
+Get device group from certificate common name
+#############################################
+
+.. code-block:: text
+
+    GET /api/v1/controller/cert/{common_name}/group/
+
+This endpoint can be used to retrieve group information and metadata by the
+common name of a certificate used in a VPN client tunnel, this endpoint is
+used in layer 2 tunneling solutions for firewall/captive portals.
+
+It is also possible to filter device group by providing organization slug
+of certificate's organization as show in the example below:
+
+.. code-block:: text
+
+    GET /api/v1/controller/cert/{common_name}/group/?org={org1_slug},{org2_slug}
+
+Get device location
+###################
+
+.. code-block:: text
+
+
+    GET /api/v1/controller/device/{id}/location/
+
+
+Create device location
+######################
+
+.. code-block:: text
+
+    POST /api/v1/controller/device/{id}/location/
+
+You can create ``DeviceLocation`` object by using primary
+keys of existing ``Location`` and ``FloorPlan`` objects as shown in
+the example below.
+
+.. code-block:: json
+
+    {
+        "location": "f0cb5762-3711-4791-95b6-c2f6656249fa",
+        "floorplan": "dfeb6724-aab4-4533-aeab-f7feb6648acd",
+        "indoor": "12.342,23.541"
+    }
+
+.. code-block:: text
+
+    curl -X POST \
+        http://127.0.0.1:8000/api/v1/controller/device/8a85cc23-bad5-4c7e-b9f4-ffe298defb5c/location/ \
+        -H 'content-type: application/json' \
+        -d '{
+            "location": "f0cb5762-3711-4791-95b6-c2f6656249fa",
+            "floorplan": "dfeb6724-aab4-4533-aeab-f7feb6648acd",
+            "indoor": "12.342,23.541"
+            }'
+
```

Comment:
let's clarify that the value in `indoor` represents the coordinates of the point placed on the image from respectively the top and left corners, a more realistic value is: `"indoor": "-279.21875,442"`, that's a value I get if I test setting a point via the UI.

Reply:
The example is missing the authorization header.